s2spy/preprocess.py

```diff
@@ -88,12 +123,56 @@ def _check_input_data(data: Union[xr.DataArray, xr.Dataset]):
         )
 
 
+def _check_temporal_resoltuion(timescale: TIMESCALE_TYPE) -> TIMESCALE_TYPE:
+    support_temporal_resolution = ["monthly", "weekly", "daily"]
+    if timescale not in support_temporal_resolution:
+        raise ValueError(
+            "Given temporal resoltuion is not supported."
+            "Please choose from 'monthly', 'weekly', 'daily'."
+        )
+    return timescale
+
+
+def _check_data_resolution_match(
+    data: Union[xr.DataArray, xr.Dataset], timescale: TIMESCALE_TYPE
+):
+    """Check if the temporal resoltuion of input is the same as given timescale."""
+    timescale_dict = {
+        "monthly": np.timedelta64(1, "M"),
+        "weekly": np.timedelta64(1, "W"),
+        "daily": np.timedelta64(1, "D"),
+    }
+    time_intervals = np.diff(data["time"].to_numpy())
+    temporal_resolution = np.median(time_intervals).astype("timedelta64[D]")
+    if timescale == "monthly":
+        temporal_resolution = temporal_resolution.astype(int)
+        if temporal_resolution <= 31 and temporal_resolution >= 28:
+            pass
+        else:
+            warnings.warn(
+                "The temporal resolution of data does not completely match "
+                "the target timescale. Please check your input data.",
+                stacklevel=1,
+            )
+
+    elif timescale in timescale_dict:
+        if timescale_dict[timescale].astype("timedelta64[D]") == temporal_resolution:
+            pass
+        else:
```

Comment:
```suggestion
        if timescale_dict[timescale].astype("timedelta64[D]") != temporal_resolution:
```

Reply:
Thanks for correcting me! I simply copy/paste the code from above and then edit it...Didn't pay attention. Looks so stupid 😅 .